### PR TITLE
Add external watchlist error and missing gray color

### PR DESCRIPTION
**Pull request by mbastos-sis**
### Description
Update the specs to match what the backend is sending, since there is a mismatch which makes the SDK crash when fetching singins in a specific case.

### Changelog
- `models/WatchlistMatch.v1.yaml`
  - Add `error` property since the backend is sending it but it was missing from the API spec.
  - Remove the required properties `name` and `list` since they're `null` when `error` is present.
- `models/ExternalWatchlistResult.v1.yaml`
  - Add missing `GRAY` color.

### CODE REVIEW

**Review by jonsangster — APPROVED** [review]

File: bundled_openapi.json [hunk 1]
```diff
@@ -5234,7 +5234,8 @@
                                   "RED",
                                   "GREEN",
                                   "YELLOW",
-                                  "ORANGE"
+                                  "ORANGE",
+                                  "GRAY"
```
Comment (ErebusBat) [inline bundled_openapi.json:5238]: Gray doesn't appear to be in our list of allowed colors...  Am I missing something?

Comment (mbastos-sis) [reply bundled_openapi.json:5238]: So this color can't be used by a user, it's only used by the backend when there was some error communicating with the external watchlist.

Comment (ErebusBat) [reply bundled_openapi.json:5238]: OH neat... I didn't know that.

**Review by ErebusBat — APPROVED** [review]

File: bundled_openapi.json (hunk 1 above)
Comment (amaury-sie) [reply bundled_openapi.json:5238]: for the record, GRAY color is set [here](https://github.com/tractionguest/guest-server/blob/5954ce73c153ee0c3f2ff2a5c3940dec4f90cde6/app/models/integration.rb#L149-L154)

**Review by amaury-sie — APPROVED** [review]

**Pull request merged**